airflow/executors/kubernetes_executor.py

```diff
@@ -216,15 +233,27 @@ def process_status(
         resource_version: str,
         event: Any,
     ) -> None:
+        annotations_string = annotations_to_str(annotations)
         """Process status response."""
         if status == "Pending":
             if event["type"] == "DELETED":
-                self.log.info("Event: Failed to start pod %s", pod_name)
+                if get_logs_task_metadata():
+                    self.log.info(
+                        "Event: Failed to start pod %s, annotations: %s", pod_name, annotations_string
+                    )
+                else:
+                    self.log.info("Event: Failed to start pod %s", pod_name)
```

Comment:
I wonder instead of doing if-else everywhere, if it’s better to simply do something like

```python
if get_logs_task_metadata():
    annotations_for_logging = annotations_to_str(annotations)
else:
    annotations_for_logging = "<omitted>"
```

and always add the `annotations: %s` part.

Reply:
Or maybe do this if-else in `annotations_to_st` directly? (If we give this function a better name.)

Reply:
Imo, leaving the omitted annotations would not look good in the logs when we do not have any annotations or the logs metadata flag enabled. What do you think?

Reply:
Logs should prioritise simplcity over making things pretty.

Reply:
Sure, let me address this comment in that case.